internal/iam/role.go

```diff
@@ -327,3 +328,34 @@ func (p *projectRole) Actions() map[string]action.Type {
 	ret[action.SetPrincipals.String()] = action.SetPrincipals
 	return ret
 }
+
+// getRoleScopeId returns scopeID for the Role from the base type iam_role table
+// use this to get scope ID to determine which of the role subtype tables to operate on
+func getRoleScopeId(ctx context.Context, r db.Reader, roleId string) (string, error) {
```

Comment:
I would consider making this function return the role's scope type, rather than the scope id. I think the common use of this will be for the repository to switch on the scope type. So perhaps make this:

```go
func getRoleScopeType(ctx context.Context, r db.Reader, roleId string) (scope.Type, error)
```